web-api/src/storage/postgres.rs

```diff
@@ -202,38 +247,43 @@ impl Database {
         tx: &mut Transaction<'_, Postgres>,
         id: &DocumentId,
     ) -> Result<bool, Error> {
-        sqlx::query("SELECT 1 FROM document WHERE document_id = $1;")
+        sqlx::query("SELECT FROM document WHERE document_id = $1;")
             .bind(id)
             .execute(tx)
             .await
             .map(|response| response.rows_affected() > 0)
             .map_err(Into::into)
     }
 
-    async fn get_interacted(
+    async fn get_snippets_for_interaction(
         tx: &mut Transaction<'_, Postgres>,
-        ids: impl IntoIterator<IntoIter = impl ExactSizeIterator<Item = &DocumentId>>,
-    ) -> Result<Vec<InteractedDocument>, Error> {
+        ids: impl IntoIterator<IntoIter = impl ExactSizeIterator<Item = &SnippetId>>,
+    ) -> Result<Vec<SnippetForInteraction>, Error> {
         let mut builder = QueryBuilder::new(
-            "SELECT document_id, tags, embedding
-            FROM document
-            WHERE document_id IN ",
+            "SELECT s.document_id, s.snippet_idx, s.embedding, d.tags
+            FROM snippet s JOIN document d USING (document_id)
+            WHERE (document_id, snippet_idx) IN ",
```

Comment:
❓ i would have assumed that we need to use the table alias as a prefix here as well, like `WHERE (d.document_id, s.snippet_idx) IN`? or does it work without the alias because that refers to the colums of the joined tables?